304 - Ids Slider add component (#326)
infor-design/enterprise-wc: src/components/ids-slider/README.md

# Ids Slider Component

## Description

> Holds the value(s) between a minimum and maximum, which can be controlled by the thumb(s) along the track of the slider. There is a single slider (1 thumb), double slider (2 thumbs), and a step slider (1 thumb and x amount of ticks). They can also be converted from horizontal (default) to vertical orientation. The value(s) can be modified by either clicking the area around the track, by dragging the thumb back and forth, or by focusing and pressing the arrow keys. 

Reviewer: ```suggestion
Displays a range of selectable values(s) between a minimum and maximum, which can be controlled by the thumb(s) along the track of the slider. There is a single slider (1 thumb), double slider (2 thumbs), and a step slider (1 thumb and x amount of ticks). They can also be converted from horizontal (default) to vertical orientation. The value(s) can be modified by either clicking the area around the track, by dragging the thumb back and forth, or by focusing and pressing the arrow keys.  A user can select a single value or range of values. 
```

## Use Cases

> - Control an input with limited options through clicking/dragging/keyboard interaction

Reviewer: ```suggestion
- Control an input with limited options through clicking/dragging/keyboard interaction.
- . Best for showing a user’s current selection relative to the lower and upper limits of a scale.
```

Author: Done

## Terminology

> - Minimum: The smallest value possible, on one end of the slider

Reviewer: ```suggestion
- **Minimum**: The smallest value possible, on one end of the slider
```

For all these please bold like the other examples

Author: Done

- Maximum: The largest value possible, on the other end of the slider
- Thumb: The circular button that can be dragged back and forth on the slider to control input
- Tick: The dots along the step slider that indicate the amount of choices for input
- Track: The bar which the thumb moves along
- Track area: The area around the track which can be clicked to modify the input value

## Settings (Attributes)

- `value` {number} set the primary value of the slider
- `value-secondary` {number} set the secondary value of the slider (applicable only for double slider)
- `min` {number} set the minimum value of the slider
- `max` {number} set the maximum value of the slider
- `color` {string} set the color of the slider
- `type` {'single' | 'double' | 'step'} set the type of the slider 
- `step-number` {number >= 2} number of steps (applicable only for the step slider)
- `vertical` {boolean}

> ## Settings (Non-attributes)

Reviewer: Whats the different with attributes and none attributes ?

Author: this if for this.labels

it's technically not an attribute because we don't want arrays being passed through HTML as strings... so they would have to set it using javascript, hence "non-attribute"

Reviewer: ok thats what i guessed. Thanks

- `labels` {array} set the list of labels you want (applicable only for step slider)

### How to set the labels

In order for the labels to be set, the `type` attribute must be set to `step`, and the attribute `step-number` must match the size of the array of labels. Javascript is required to inject the label array into the DOM.

```js

> this.labels = ['very bad, poor, average, good, excellent']; // where 'this' is the ids-slider

Reviewer: So there is no way to set the label in the DOM? Or is that just for the step ones because its an object/array

Author: Yeah this is for the step slider... currently the default labels are the min and max value
